### ISSUE TITLE
Cannot run `@report_call` on Julia 1.8-beta1

### ISSUE THREAD

**Issue by jakobnissen** [aviatesk/JET.jl#326]
This is using Julia 1.8.0-beta1, in an empty environment with JET 0.5.7:

The reproducer (same as #321 ) and top of stacktrace
```
julia> using JET

julia> function f()
           v = map((2,)) do i
              isodd(i) ? i : ""
           end
           convert(Tuple{Int}, v)
       end
f (generic function with 1 method)

julia> @report_call f()
ERROR: TypeError: in typeassert, expected Pair{Symbol, Core.Compiler.InferenceResult}, got a value of typ
e Nothing
Stacktrace:
  [1] cache_lookup(linfo::Core.MethodInstance, given_argtypes::Vector{Any}, cache::JET.JETLocalCache{JET.
JETAnalyzer{JET.BasicPass{typeof(JET.basic_function_filter)}}})
    @ JET ~/.julia/packages/JET/pV0zf/src/abstractinterpret/typeinfer.jl:497
  [2] abstract_call_method_with_const_args(interp::JET.JETAnalyzer{JET.BasicPass{typeof(JET.basic_functio
n_filter)}}, result::Core.Compiler.MethodCallResult, f::Any, arginfo::Core.Compiler.ArgInfo, match::Core.
MethodMatch, sv::Core.Compiler.InferenceState)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:757
  [3] abstract_call_gf_by_type(interp::JET.JETAnalyzer{JET.BasicPass{typeof(JET.basic_function_filter)}},
 f::Any, arginfo::Core.Compiler.ArgInfo, atype::Any, sv::Core.Compiler.InferenceState, max_methods::Int64
)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:139
  [4] abstract_call_gf_by_type
    @ ~/.julia/packages/JET/pV0zf/src/analyzers/jetanalyzer.jl:319 [inlined]
  [5] abstract_call_known(interp::JET.JETAnalyzer{JET.BasicPass{typeof(JET.basic_function_filter)}}, f::A
ny, arginfo::Core.Compiler.ArgInfo, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1633
  [6] abstract_call(interp::JET.JETAnalyzer{JET.BasicPass{typeof(JET.basic_function_filter)}}, arginfo::C
ore.Compiler.ArgInfo, sv::Core.Compiler.InferenceState, max_methods::Int64)
    @ Core.Compiler ./compiler/abstractinterpretation.jl:1691
```

**Comment by dpinol** [comment]
I can also reproduce it with 
```
using Random

julia> @report_call Random.seed!(42)
```
Or with `report_package`

**Comment by aviatesk** [comment]
fixed by efe7aa86458f03d75393587e4fe92ee937e8a0ff and on v0.5.8
